mkosi/gentoo.py

```diff
@@ -0,0 +1,462 @@
+import multiprocessing
+import os
+import re
+import tarfile
+import urllib.parse
+import urllib.request
+from pathlib import Path
+from textwrap import dedent
+from typing import Dict, List, Sequence
+
+from .backend import (
+    ARG_DEBUG,
+    CommandLineArguments,
+    MkosiPrinter,
+    OutputFormat,
+    die,
+    run_workspace_command,
+)
+
+NO_PORTAGE_DIE_MSG = "You need portage(5) - the heart of Gentoo"
+PORTAGE_GIT_URL = "https://gitweb.gentoo.org/proj/portage.git"
+try:
+    from portage.const import (  # type: ignore
+        CUSTOM_PROFILE_PATH,
+        EBUILD_SH_ENV_DIR,
+        PROFILE_PATH,
+        USER_CONFIG_PATH,
+    )
+    from portage.package.ebuild.config import config  # type: ignore
+except ImportError:
+    die(f"{NO_PORTAGE_DIE_MSG} : {PORTAGE_GIT_URL}")
```

Comment:
After I installed this using pip, I didn't have PYTHONPATH set properly under sudo, and the error was rather frustrating. I think this needs to be more precise: please print the import error, and only then do the `die` message.

Also, the upstream git repo is neither very welcoming, nor very useful. Wouldn't it be better to link to pypi, https://pypi.org/project/portage/, and maybe even mention `pip3 user --install portage` ?

Reply:
it now prints the same steps we use in `ci.yml` since it's supposedly works on any systemd-based machine (works on ubuntu on github).

i can't exactly reproduce here, (can't see what happens to `raise MkosiException(e)`,  please verify.